pkg/controller/secret/secret_controller_test.go

```diff
@@ -254,7 +267,42 @@ func verifyInhibitRules(t *testing.T, inhibitRules []*alertmanager.InhibitRule)
 				"alertname": "KubeDaemonSetRolloutStuck",
 			},
 			Equal: []string{
-				"namespace",
+				"instance",
+			},
+			Expected: true,
+		},
+		{
+			SourceMatch: map[string]string{
+				"alertname": "KubeNodeNotReady",
+			},
+			TargetMatchRE: map[string]string{
+				"alertname": "KubeDaemonSetMisScheduled",
+			},
+			Equal: []string{
+				"instance",
+			},
+			Expected: true,
+		},
+		{
+			SourceMatch: map[string]string{
+				"alertname": "KubeNodeNotReady",
+			},
+			TargetMatchRE: map[string]string{
+				"alertname": "SDNPodNotReady",
+			},
+			Equal: []string{
+				"instance",
+			},
+			Expected: true,
+		},
+		{
+			SourceMatch: map[string]string{
+				"alertname": "KubeNodeNotReady",
+			},
+			TargetMatchRE: map[string]string{
+				"alertname": "TargetDown",
```

Comment:
Should that really fire if we drop a node?

Reply:
It does, yes. TargetDown is if a service is missing pods. If a node is down, then the pods will be removed from the service.

Reply:
Isn't that only for single service pods or all pods of a service on a single node?

Reply:
Sorry, I don't quite understand your question.

Reply:
Sorry for not being clear.
A target should normally be a service and that service should have multiple backends, namely pods.
So the alert for target down should only fire if all pods of a given service are not reacting to the prometheus request.
That can technically only happen when a node is unresponsive if all pods of one service are on the same node or that service only has one pod attached.

Reply:
The expression for TargetDown is:
```
100 * (count by(job, namespace, service) (up == 0) / count by(job, namespace, service) (up)) > 10
```

So if more than 10% of the pods in a service are down, then it will fire.

Reply:
thanks for looking that up.